### PR TITLE
Add client post object

### PR DESCRIPTION
**Pull request by kyle-fritz-zocdoc**
@ZocDoc/coreinfrastructure 

This PR implements ServiceClientBase.Post(object) and PostAsync(object).

'object' is better for ZocDoc because we don't want to rely on the ServiceStack IReturn interface needed by the existing Post/PostAsync methods.

(David C. is the actual author, please disregard the kyle-fritz-zocdoc author mentioned above)

### CODE REVIEW

**Comment by ben-sussman-zocdoc** [comment]
Sweet :+1: It might make sense to post a code snippet of our client wrapper in this PR to show ppl how we are doing the strong typing in our wrapper instead of requiring a reference to ServiceStack in all of our contract projects. 

Wrappers are likely common for users of ServiceStack as the figuring out of the endpoint base url should be something automatic and not manually specified by every caller.

**Comment by david-chenevert-zocdoc** [comment]
The ServiceStack libraries are in a fork of ServiceStack code base, version 3.
It adds ServiceClientBase.Post/Async(object,...) and ServiceClientBase.PostAsync(object,...), so that DTO definitions do not depend on ServiceStack.Interfaces

File: src/ServiceStack.Common/ServiceClient.Web/JsonRestClientAsync.cs [hunk 1]
```diff
@@ -82,6 +82,11 @@ public void PostAsync<TResponse>(IReturn<TResponse> request, Action<TResponse> o
             throw new NotImplementedException();
         }
 
+        public void PostAsync<TResponse>(object request, Action<TResponse> onSuccess, Action<TResponse, Exception> onError)
+        {
+            throw new NotImplementedException();
```
Comment (kyle-fritz-zocdoc) [inline src/ServiceStack.Common/ServiceClient.Web/JsonRestClientAsync.cs:87]: Not implemented?

Comment (david-chenevert-zocdoc) [inline src/ServiceStack.Common/ServiceClient.Web/JsonRestClientAsync.cs:87]: There are several 'clients' in SS. The one we use is named JsonServiceClient. We don't use this one but it needs to build, so I added this method.

**Pull request merged**